fortio_main.go

```diff
@@ -536,9 +537,23 @@ func grpcClient() {
 		}
 		return
 	}
-	_, err := fgrpc.PingClientCall(host, count, httpOpts.PayloadUTF8(), *pingDelayFlag, &httpOpts.TLSOptions)
+	_, err := fgrpc.PingClientCall(host, count, httpOpts.PayloadUTF8(), *pingDelayFlag, &httpOpts.TLSOptions, grpcMD())
 	if err != nil {
 		// already logged
 		os.Exit(1)
 	}
 }
+
+func grpcMD() map[string][]string {
```

Comment:
could be `httpHeader2grpcMeta(headers http.Header) {` ? but... what's wrong with just using ExtraHeaders directly ? they are both a `map[string][]string` aren't they?

Reply:
First, all keys are converted to lowercase here. This is a requirement of grpc-go, otherwise there will be an error.
https://github.com/grpc/grpc-go/blob/f2fbb0e07ebf3dd46aa641ee89c9f17e8083eaf6/stream.go#L170-L175
https://github.com/grpc/grpc-go/blob/f2fbb0e07ebf3dd46aa641ee89c9f17e8083eaf6/internal/metadata/metadata.go#L94-L96

Second, the processing of the `user-agent` is skipped here, just to leave this todo to identify that there is some work to be done here: grpc can only set HTTP standard headers like `user-agent`, `:authority`, etc. by passing `DialOption` to the `Dial()` , and considering the current code structure, the amount of changes to implement the conversion of these standard headers may be a bit large, so I thought it would be better to do it in another MR.
`httpHeader2grpcMeta` is indeed a better name, and we should adopt it.

Reply:
It's a bit strange here, the header validation will only work for headers set using the grpc.NewOutgoingContext method, which I happened to use (instead of grpc.AppendToOutgoingContext)
The second return value is ignored:
```
If md, _, ok := metadata.FromOutgoingContextRaw(ctx); ok {
```
But I think at least the lowercase conversion here is harmless